kzg_prover/src/circuits/tests.rs

```diff
@@ -122,39 +122,64 @@ mod test {
         // Both the Custodian and the Verifier know what column range are the balance columns
         let balance_column_range = 1..N_CURRENCIES + 1;
 
-        // The Custodian communicates the KZG opening transcripts to the Verifier
-        // The Verifier verifies the KZG opening transcripts and calculates the grand sums
+        // The Custodian communicates the KZG multi-opening transcript to the Verifier
+        // The Verifier verifies the KZG multi-opening and calculates the grand sums
         let (verified, grand_sum) = verify_grand_sum_openings::<N_CURRENCIES>(
             &params,
             &zk_snark_proof,
-            kzg_proofs,
+            grand_sums_multi_proof,
             poly_degree,
             balance_column_range,
         );
 
+        assert!(verified);
         for i in 0..N_CURRENCIES {
-            assert!(verified[i]);
             assert_eq!(csv_total[i], grand_sum[i]);
         }
 
-        let balance_column_range = 1..N_CURRENCIES + 1;
-        let (balances_verified, balance_values) = verify_user_inclusion::<N_CURRENCIES>(
+        let column_range = 0..N_CURRENCIES + 1;
+        // The Verifier verifies the inclusion of the 4th user balances
+        const N_POINTS: usize = N_CURRENCIES + 1;
+        let (inclusion_verified, id_and_balance_values) = verify_user_inclusion::<N_POINTS>(
             &params,
             &zk_snark_proof,
-            balance_opening_proofs,
-            balance_column_range,
+            &openings_multi_proof,
+            column_range,
             omega,
             user_index,
         );
 
+        assert!(inclusion_verified);
         let fourth_user_csv_entry = entries.get(user_index as usize).unwrap();
-        for i in 0..N_CURRENCIES {
-            assert!(balances_verified[i]);
-            assert_eq!(
-                *fourth_user_csv_entry.balances().get(i).unwrap(),
-                balance_values[i]
-            );
+        for i in 0..N_CURRENCIES + 1 {
+            if i == 0 {
+                assert_eq!(
+                    *fourth_user_csv_entry.username_as_big_uint(),
+                    id_and_balance_values[i]
+                );
+            } else {
+                assert_eq!(
+                    *fourth_user_csv_entry.balances().get(i - 1).unwrap(),
+                    id_and_balance_values[i]
+                );
+            }
         }
+
+        let balance_column_range = 1..N_CURRENCIES + 1;
+
+        // Test failure case with the wrong group generator
```

Comment:
I suggest to separate the test failure in a different test vector. Also probably it makes sense to add more failure test and assert their specific unexpected behaviour.

Reply:
Will move it into a separate negative test and create an issue for the additional negative tests.

Reply:
https://github.com/summa-dev/summa-solvency/issues/229